app/src/main/java/com/realityexpander/tasky/agenda_feature/data/repositories/eventRepository/eventRepositoryImpls/EventRepositoryImpl.kt

```diff
@@ -7,28 +7,43 @@ import com.realityexpander.tasky.agenda_feature.data.common.convertersDTOEntityD
 import com.realityexpander.tasky.agenda_feature.data.common.convertersDTOEntityDomain.toEventDTOUpdate
 import com.realityexpander.tasky.agenda_feature.data.repositories.eventRepository.local.IEventDao
 import com.realityexpander.tasky.agenda_feature.data.repositories.eventRepository.remote.eventApi.IEventApi
+import com.realityexpander.tasky.agenda_feature.data.repositories.syncRepository.ISyncRepository
 import com.realityexpander.tasky.agenda_feature.domain.AgendaItem
 import com.realityexpander.tasky.agenda_feature.domain.IEventRepository
 import com.realityexpander.tasky.agenda_feature.domain.ResultUiText
+import com.realityexpander.tasky.auth_feature.domain.IAuthRepository
 import com.realityexpander.tasky.core.presentation.common.util.UiText
 import com.realityexpander.tasky.core.util.rethrowIfCancellation
 import kotlinx.coroutines.flow.Flow
 import kotlinx.coroutines.flow.map
 import java.time.ZonedDateTime
 
 class EventRepositoryImpl(
-    private val eventDao: IEventDao, // = EventDaoFakeImpl(),
-    private val eventApi: IEventApi, // = EventApiFakeImpl(),
+    private val eventDao: IEventDao,
+    private val eventApi: IEventApi,
+    private val syncRepository: ISyncRepository,
+    private val authRepository: IAuthRepository  // todo add to DI constructor
 ) : IEventRepository {
 
     // • CREATE
 
-    override suspend fun createEvent(event: AgendaItem.Event): ResultUiText<AgendaItem.Event> {
+    override suspend fun createEvent(event: AgendaItem.Event, isRemoteOnly: Boolean): ResultUiText<AgendaItem.Event> {
         return try {
-            eventDao.createEvent(event.toEntity())  // save to local DB first
+            if(!isRemoteOnly) {
+                eventDao.createEvent(event.copy(isSynced = false).toEntity())  // save to local DB first
+                syncRepository.addCreatedSyncItem(event)
+            }
```

Comment:
In which case would you not want to save it locally?

Reply:
When the app is syncing with the server, i only want to `create` an item on the server, and not locally (the app already has the item locally).

When the app is online, it will attempt to save to the server during `create`.

Is there something I'm missing here?

Reply:
Ah okay, yeah you can do it like that. Feels a bit of a hassle to me to save one DB call while adding extra params and so, but it's fine

Reply:
Yeah I considered making separate calls, but it was a toss up.